sqlcapture/capture.go

```diff
@@ -648,27 +648,30 @@ func (c *Capture) backfillStream(ctx context.Context, streamID string, discovery
 }
 
 func (c *Capture) emitChange(event *ChangeEvent) error {
-	var record map[string]interface{}
+	var record = make(map[string]any)
 	var meta = struct {
-		Operation ChangeOp               `json:"op"`
-		Source    SourceMetadata         `json:"source"`
-		Before    map[string]interface{} `json:"before,omitempty"`
+		Operation ChangeOp       `json:"op"`
+		Source    SourceMetadata `json:"source"`
+		Before    map[string]any `json:"before,omitempty"`
 	}{
 		Operation: event.Operation,
 		Source:    event.Source,
 		Before:    nil,
 	}
 	switch event.Operation {
 	case InsertOp:
-		record = event.After // Before is never used.
+		for k, v := range event.After {
```

Comment:
what motivated the extra allocation / copy here?

Reply:
Previously I had the key-change logic emitting a deletion event using the 'before' row state, and then an update with the 'before' state in the metadata, and the in-place modification of this map resulted in an extra `_meta` field getting embedded into that before-state metadata.

After changing the logic to omit any before-state metadata and just treat the creation as a pure insert-from-nothing I think it would be safe to revert this fix if we think the additional allocation is a significant bottleneck.

Reply:
I've reverted the sqlcapture memory-reuse change since it's no longer necessary for correctness after all, so now the changes in this PR (other than the new tests for Postgres and SQL Server) are isolated to just MySQL.